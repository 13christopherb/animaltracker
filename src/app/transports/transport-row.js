import React, {Component} from "react";
import moment from 'moment';


export const TransportRow = (props) => {

    const deleteTransport = (e) => {
        e.preventDefault();
        props.deleteTransport(props.transport)
    };

    return (
        <tr key={props.transport.id}>
            <td style={{width: '16.66%'}}>
                {props.transport.departs}
            </td>
            <td style={{width: '16.66%'}}>
                {props.transport.arrives}
            </td>
            <td style={{width: '16.66%'}}>
                {moment(props.transport.meetTime).format('dddd MM/DD hhmm')}
            </td>
            <td style={{width: '16.66%'}}>
                <button onClick={deleteTransport} id="delete" className="btn btn-danger">Delete</button>
            </td>
        </tr>
    )
};