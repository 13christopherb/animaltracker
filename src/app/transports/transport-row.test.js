import React from 'react';
import {shallow} from 'enzyme';
import {TransportRow} from "./transport-row";
import moment from 'moment';
import toJson from 'enzyme-to-json';

const transport = {
    departs: 'MBO',
    arrives: 'NRO',
    meetTime: moment('2017-09-15 09:30:00') //Arbitrary time
};

describe('<TransportRow />', () => {

    describe('render()', () => {
        test('should render the component', () => {
            const wrapper = shallow(<TransportRow transport={transport}/>);

            expect(toJson(wrapper)).toMatchSnapshot()
        })
    });
    describe('Buttons', () => {
        test('delete button calls deleteAnimal', () => {
            const mockDelete = jest.fn();
            const wrapper = shallow(
                <TransportRow deleteTransport={mockDelete} transport={transport}/>
            );

            wrapper.find('#delete').simulate('click', { preventDefault() {} });
            expect(mockDelete).toHaveBeenCalled()
        });
    });
});