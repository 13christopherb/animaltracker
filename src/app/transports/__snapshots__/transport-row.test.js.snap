// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TransportRow /> render() should render the component 1`] = `
<tr>
  <td
    style={
      Object {
        "width": "16.66%",
      }
    }
  >
    MBO
  </td>
  <td
    style={
      Object {
        "width": "16.66%",
      }
    }
  >
    NRO
  </td>
  <td
    style={
      Object {
        "width": "16.66%",
      }
    }
  />
  <td
    style={
      Object {
        "width": "16.66%",
      }
    }
  >
    Friday 09/15 0930
  </td>
  <td
    style={
      Object {
        "width": "16.66%",
      }
    }
  >
    <ForwardRef(Bootstrap(Button))
      id="delete"
      onClick={[Function]}
      variant="danger"
    >
      Delete
    </ForwardRef(Bootstrap(Button))>
  </td>
</tr>
`;
